src/content/docs/apm/agents/ruby-agent/instrumented-gems/grpc-instrumentation.mdx

```diff
@@ -0,0 +1,119 @@
+
+<Callout variant="tip">
+  By default, the New Relic Ruby agent has instrumentation for both gRPC clients and servers set to `auto`.
+
+  By modifying the New Relic config file or by setting environment variables, the agent's gRPC instrumentation behavior can be altered with one of the following values:
+
+  - **auto:** the default, enabled via automatic Ruby method behavior selection
+  - **disabled:** instrumentation disabled
+  - **prepend:** force the use of Ruby method prepending
+  - **chain:** force the use of Ruby method chaining
+
+  **gRPC client**
+  - config file parameter: `instrumentation.grpc_client`
+  - environment variable: `NEW_RELIC_INSTRUMENTATION_GRPC_CLIENT`
+
+  **gRPC server**
+  - config file parameter: `instrumentation.grpc_server`
+  - environment variable: `NEW_RELIC_INSTRUMENTATION_GRPC_SERVER`
+
+  Examples:
+
+  To disable gRPC server instrumentation via the config file:
+
+  ```
+  # newrelic.yml
+  instrumentation.grpc_server: disabled
+  ```
+
+  To disable the gRPC client instrumentation via an environment variable:
+
+  ```
+  export NEW_RELIC_INSTRUMENTATION_GRPC_CLIENT=disabled
+  ```
+</Callout>
+
+
+## Verification
+
+Once the New Relic Ruby agent starts, it will detect the presence of the `grpc` gem having been loaded and then attempt to apply gRPC instrumentation. When successful at applying this instrumentation for the gRPC client and server components, the agent will write the following lines to its log file (`log/newrelic_agent.log` by default):
+
+```
+INFO : Installing New Relic supported gRPC_Client instrumentation using Prepend
+INFO : Installing New Relic supported gRPC_Server instrumentation using Prepend
+```
+
+
+## Usage
+
+As long as the instrumentation is working properly (see "Verification" above to confirm this), all instrumentation performed by the New Relic Ruby agent should be fully automatic and not require any modifications whatsoever to any of your existing gRPC client and/or server code.
+
+The New Relic Ruby agent's gRPC instrumentation can be used for New Relic Distributed Tracing in conjunction with any other New Relic agent that supports gRPC. gRPC requests can be sent from a Ruby-based client to a server based on another language, and a Ruby-based server can field requests coming from a client based on another language.
+
+<Callout variant="important">
+  gRPC client requests will not be reported to New Relic unless they are spawned when an existing New Relic transaction is present. Typically this means that a Rails controller method or background job process is active and responsible for the invocation of the gRPC client request. The request will then roll up under that outer wrapper's transaction in the New Relic web UI.
+
+  gRPC server request handlings will always generate a new New Relic transaction and will not require nor roll-up to any pre-existing New Relic transaction.
```

Comment:
```suggestion
  gRPC server requests will always generate a new New Relic transaction and will not require nor roll-up to any pre-existing New Relic transaction.
```

Reply:
"handlings" was a deliberate term to incorporate streaming (one long running open connection with multiple payloads across what from an HTTP perspective might be a single "request"), but this approach doesn't require such a lengthy footnote, merging.